code/modules/research/research.dm

```diff
@@ -91,14 +91,14 @@ research holder datum.
 	return
 
 /datum/research/proc/AddDesign2Known(var/datum/design/D)
-	if(!known_designs.len) // Special case
+	if(!length(known_designs))
 		known_designs.Add(D)
 		return
 	for(var/i = 1 to known_designs.len)
 		var/datum/design/A = known_designs[i]
-		if(A.id == D.id) // We are guaranteed to reach this if the ids are the same, because sort_string will also be the same
+		if(A.name == D.name)
 			return
-		if(A.sort_string > D.sort_string)
+		if("[A.name]" > "[D.name]")
```

Comment:
Doesn't `if(A.name > D.name)` work?
Seems a little odd to create a new string out of what should already be a string

Reply:
It runtimes on a null - I actually had name initialize and forgot to reverse this.